server/lobbyconnection.py

```diff
@@ -786,38 +790,45 @@ def command_game_host(self, message):
         mapname = message.get('mapname')
         password = message.get('password')
 
-        game = self.game_service.create_game(**{
-            'visibility': visibility,
-            'game_mode': mod.lower(),
-            'host': self.player,
-            'name': title if title else self.player.login,
-            'mapname': mapname,
-            'password': password
-        })
-        self.launch_game(game, port, True)
+        game_mode = GameMode.from_string(mod.lower())
+
+        game = self.game_service.create_game(
+            visibility=visibility,
+            game_mode=game_mode,
+            host=self.player,
+            name=title if title else self.player.login,
+            mapname=mapname,
+            password=password
+        )
+        self.launch_game(game, port, is_host=True)
         server.stats.incr('game.hosted')
 
     def launch_game(self, game, port, is_host=False, use_map=None):
-        # FIXME: Setting up a ridiculous amount of cyclic pointers here
+        # TODO: Fix setting up a ridiculous amount of cyclic pointers here
         if self.game_connection:
             self.game_connection.abort("Player launched a new game")
-        self.game_connection = GameConnection(self.loop,
-                                              self,
-                                              self.player_service,
-                                              self.game_service)
-        self.game_connection.player = self.player
-        self.player.game_connection = self.game_connection
```

Comment:
Need to add this back in

Reply:
Probably best to do it in the constructor of GameConnection since that constructor now takes the player as an argument anyways.